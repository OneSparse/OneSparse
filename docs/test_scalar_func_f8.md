# Scalar Functions

This documentation is also tests for the code, the examples below
show the literal output of these statements from Postgres.

Some setup to make sure warnings are shown, and that the extension
is installed.
```
set client_min_messages = 'WARNING';
create extension if not exists onesparse;

```
Test construction of min, zero and max values:
```
select '-1'::float8::scalar;
select '0'::float8::scalar;
select '1'::float8::scalar;

```
Test setting a scalar value from max to 2
```
select set('1'::float8::scalar, 2);

```
Test various casting functions used by the CREATE CAST machinery:
```
select scalar_float8((-1)::float8);
select scalar_float8((0)::float8);
select scalar_float8((1)::float8);
select float8_scalar((-1)::float8::scalar);
select float8_scalar((0)::float8::scalar);
select float8_scalar((1)::float8::scalar);
```