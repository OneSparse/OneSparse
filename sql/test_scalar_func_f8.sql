set client_min_messages = 'WARNING';
create extension if not exists onesparse;
set search_path = 'onesparse';
select '-1'::float8::scalar;
select '0'::float8::scalar;
select '1'::float8::scalar;
select set('1'::float8::scalar, 2);
select scalar_float8((-1)::float8);
select scalar_float8((0)::float8);
select scalar_float8((1)::float8);
select float8_scalar((-1)::float8::scalar);
select float8_scalar((0)::float8::scalar);
select float8_scalar((1)::float8::scalar);
