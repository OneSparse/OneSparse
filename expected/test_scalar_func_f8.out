-- # Scalar 'float8' Functions
--
-- This documentation is also tests for the code, the examples below
-- show the literal output of these statements from Postgres.
--
-- Some setup to make sure warnings are shown, and that the extension
-- is installed.
set client_min_messages = 'WARNING';
create extension if not exists onesparse;
-- Test construction of min, zero and max values:
select '-1'::float8::scalar;
    scalar    
--------------
 f8:-1.000000
(1 row)

select '0'::float8::scalar;
   scalar    
-------------
 f8:0.000000
(1 row)

select '1'::float8::scalar;
   scalar    
-------------
 f8:1.000000
(1 row)

-- Test setting a scalar value from max to 2
select set('1'::float8::scalar, 2);
     set     
-------------
 f8:2.000000
(1 row)

-- Test various casting functions used by the CREATE CAST machinery:
select scalar_float8((-1)::float8);
 scalar_float8 
---------------
 f8:-1.000000
(1 row)

select scalar_float8((0)::float8);
 scalar_float8 
---------------
 f8:0.000000
(1 row)

select scalar_float8((1)::float8);
 scalar_float8 
---------------
 f8:1.000000
(1 row)

select float8_scalar((-1)::float8::scalar);
 float8_scalar 
---------------
            -1
(1 row)

select float8_scalar((0)::float8::scalar);
 float8_scalar 
---------------
             0
(1 row)

select float8_scalar((1)::float8::scalar);
 float8_scalar 
---------------
             1
(1 row)

